regression-test/framework/src/main/groovy/org/apache/doris/regression/util/JdbcUtils.groovy

```diff
@@ -18,37 +18,40 @@
 package org.apache.doris.regression.util
 
 import com.google.common.collect.ImmutableList
+import org.apache.doris.regression.util.Metaholder
 
 import java.sql.Connection
 import java.sql.ResultSet
+import java.sql.ResultSetMetaData
 
 class JdbcUtils {
-    static List<List<Object>> executeToList(Connection conn, String sql) {
+    static List<List<Object>> executeToList(Connection conn, String sql, Metaholder holder) {
         conn.prepareStatement(sql).withCloseable { stmt ->
```

Comment:
suggest `new Metaholder` in `JdbcUtils` and return, and you can extract the interest metadata out of the `Connection` life cycle safely.

e.g.

```groovy
class Metadata {
  public final List<String> columnNames;
  public final List<String> columnTypes;
}

static Tuple2<List<List<Object>>, Metadata> executeToList(Connection conn, String sql) {
  def metadata = new Metadata()
  resultList = ...
  metadata.columnNames = ...
  metadata.columnTypes = ...
  return [resultList, metadata]
}
```
then, you can take result by this code
```groovy
def (result, metadata) = JdbcUtils.executeToList(connnection, sql)
```

Reply:
fixed